torchvision/datasets/kinetics.py

```diff
@@ -74,6 +150,106 @@ def __init__(self, root, frames_per_clip, step_between_clips=1, frame_rate=None,
         )
         self.transform = transform
 
+    def _set_up_paths(self, root, split) -> None:
+        """Sets up self.root and self.split to avoid confusion.
+        Split in the root (e.g. kinetics/val) overrides the setting in 
+        the split.
+        """
+        self.split = split
+        if path.basename(root) == split:
+            self.root = root
+        elif path.basename(root) in ["train", "val"]:
+            self.root = root
+            self.split = path.basename(root)
+            warnings.warn(
+                f"Root {root} points to a different split than {split}."
+                f"Assigning self.split to {self.split}."
+            )
+        else:
+            self.root = path.join(root, split)
+
+    def download_and_process_videos(self) -> None:
+        """
+        downloads all the videos to the _root_ folder
+        in the expected format
+        """
+        tic = time.time()
+        _ = self._download_videos()
+        toc = time.time()
+        print("Elapsed time for downloading in mins ", (toc - tic) / 60)
+        self._make_ds_structure()
+        toc2 = time.time()
+        print("Elapsed time for processing in mins ", (toc2 - toc) / 60)
+        print("Elapsed time overall in mins ", (toc2 - tic) / 60)
+
+    def _download_videos(self) -> int:
+        """download tarballs containing the video to 
+        "tars" folder and extract them into the _split_ folder
+        where split is one of the official dataset splits.
+
+        Raises:
+            RuntimeError: if download folder exists, break to prevent
+              downloading entire dataset again.
+        """
+        if path.exists(self.root):
+            raise RuntimeError(
+                f"The directory {self.root} already exists. If you want to re-download or re-extract the images, "
+                f"delete the directory."
+            )
+
+        file_url = urllib.request.urlopen(
+            self._FILES[self.n_classes].format(split=self.split)
+        )
+        kinetics_dir, _ = path.split(self.root)
+        tar_path = path.join(kinetics_dir, "tars")
+
+        if self._num_download_workers < 2:
```

Comment:
Maybe turn this block around to make the intent more clear:

```suggestion
        if self._num_download_workers > 1:
```